impl/src/main/java/org/jboss/weld/bean/proxy/ProxyFactory.java

```diff
@@ -911,4 +918,17 @@ protected void getMethodHandlerField(ClassFile file, CodeAttribute b) {
     protected Class<?> getProxySuperclass() {
         return getBeanType().isInterface() ? Object.class : getBeanType();
     }
+
+    private static String getDefaultPackageReason(Class<?> clazz) {
+        if (clazz.getPackage() == null) {
+            return NULL;
+        }
+        if (clazz.getSigners() != null) {
+            return SIGNED;
+        }
+        if (clazz.getPackage().isSealed()) {
```

Comment:
I think we need not tackle this for sealed packages. Last time I tried this was working just fine - I will try and add an automatic test for sealed packages (should be doable with Shrinkwrap).

The reason why I think it works is that sealed packages are checked once while checking the classpath (verifies that given package is present in at most one JAR) and you have no restrictions on adding stuff later on.